easybuild/easyconfigs/k/kaggle/kaggle-1.6.17-foss-2023b.eb

```
@@ -0,0 +1,48 @@
+easyblock = 'PythonBundle'
+
+name = 'kaggle'
+version = '1.6.17'
+
+homepage = "https://github.com/Kaggle/kaggle-api"
+description = """Official API for https://www.kaggle.com,
+ accessible using a command line tool implemented in Python 3."""
+
+toolchain = {'name': 'foss', 'version': '2023b'}
+
+builddependencies = [
+    ('hatchling', '1.18.0'),
+]
+
+dependencies = [
+    ('Python', '3.11.5'),
+    ('tqdm', '4.66.2'),
+    # for certifi, python-dateutil, requests, six, urllib3 and webencodings
+    ('Python-bundle-PyPI', '2023.10'),
+]
+
+use_pip = True
+sanity_pip_check = True
+
+exts_list = [
+    ('bleach', '6.2.0', {
+        'source_tmpl': SOURCE_PY3_WHL,
+        'checksums': ['117d9c6097a7c3d22fd578fcd8d35ff1e125df6736f554da4e432fdd63f31e5e'],
+    }),
+    ('text_unidecode', '1.3', {
+        'source_tmpl': SOURCE_WHL,
+        'checksums': ['1311f10e8b895935241623731c2ba64f4c455287888b18189350b67134a822e8'],
+    }),
+    ('python_slugify', '8.0.4', {
+        'modulename': 'slugify',
+        'source_tmpl': SOURCE_WHL,
+        'checksums': ['276540b79961052b66b7d116620b36518847f52d5fd9e3a70164fc8c50faa6b8'],
+    }),
+    (name, version, {
+        # `import kaggle` fails for missing API token (`kaggle.json`)
+        # see `api.authenticate()` in `kaggle/__init__.py`
+        'modulename': False,
```

As discussed, just try importing `kaggle.api` instead?

```suggestion
        'modulename': 'kaggle.api',

---

That doesn't help because either way the `kaggle/__init__.py` gets executed.
```
== 2024-11-18 09:28:29,815 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/easybuild-framework/easybuild/base/exceptions.py:126 in __init__): Sanity check failed: extensions sanity check failed for 1 extensions: kaggle
failing sanity check for 'kaggle' extension: command "python -c "import kaggle.api"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/cascadelake-ampere-ib/software/kaggle/1.6.17-foss-2023b/lib/python3.11/site-packages/kaggle/__init__.py", line 7, in <module>
    api.authenticate()
  File "/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/cascadelake-ampere-ib/software/kaggle/1.6.17-foss-2023b/lib/python3.11/site-packages/kaggle/api/kaggle_api_extended.py", line 407, in authenticate
    raise IOError('Could not find {}. Make sure it\'s located in'
OSError: Could not find kaggle.json. Make sure it's located in /user/gent/453/vsc45304/.config/kaggle. Or use the environment method. See setup instructions at https://github.com/Kaggle/kaggle-api/,  (at easybuild/easybuild-framework/easybuild/framework/easyblock.py:3670 in _sanity_check_step)
```

---

Alright, then let's just go with `False`, and keep the comment